conseil-common/src/main/scala/tech/cryptonomic/conseil/common/mantis/MantisPersistence.scala

```diff
@@ -0,0 +1,260 @@
+import tech.cryptonomic.conseil.common.mantis.rpc.json.{Block, Log, Transaction}
+import tech.cryptonomic.conseil.common.mantis.MantisPersistence._
+import tech.cryptonomic.conseil.common.mantis.rpc.json.TransactionReceipt
+import tech.cryptonomic.conseil.common.mantis.domain.{Contract, Token}
+
+/**
+  * Mantis persistence into the database using Slick.
+  */
+class MantisPersistence[F[_]: Concurrent] extends LazyLogging {
+
+  /**
+    * SHA-3 signature for: Transfer(address,address,uint256)
+    */
+  private val tokenTransferSignature = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"
+
+  /**
+    * Create [[DBIO]] seq with blocks and transactions that can be wrap into one transaction.
+    *
+    * @param block JSON_RPC block
+    * @param transactions JSON_RPC block's transactions
+    */
+  def createBlock(
+      block: Block,
+      transactions: List[Transaction],
+      receipts: List[TransactionReceipt]
+  ): DBIOAction[Unit, NoStream, Effect.Write] =
+    DBIO.seq(
+      Tables.Blocks += block.convertTo[Tables.BlocksRow],
+      Tables.Transactions ++= transactions.map(_.convertTo[Tables.TransactionsRow]),
+      Tables.Receipts ++= receipts.map(_.convertTo[Tables.ReceiptsRow]),
+      Tables.Logs ++= receipts.flatMap(_.logs).map(_.convertTo[Tables.LogsRow]),
+      Tables.TokenTransfers ++= receipts
+            .flatMap(_.logs)
+            .filter(
+              log =>
+                log.topics.size == 3 && log.topics
+                    .contains(tokenTransferSignature)
+            )
+            .map(_.convertTo[Tables.TokenTransfersRow])
+    )
+
+  /**
+    * Create [[DBIO]] seq with contracts.
+    *
+    * @param logs JSON_RPC contract
+    */
+  def createContracts(contracts: List[Contract]) =
+    DBIO.seq(
+      Tables.Contracts ++= contracts.map(_.convertTo[Tables.ContractsRow])
+    )
+
+  /**
+    * Create [[DBIO]] seq with tokens.
+    *
+    * @param logs JSON_RPC token
+    */
+  def createTokens(tokens: List[Token]) =
+    DBIO.seq(
+      Tables.Tokens ++= tokens.map(_.convertTo[Tables.TokensRow])
+    )
+
+  /**
+    * Get sequence of existing blocks heights from the database.
+    *
+    * @param range Inclusive range of the block's height
+    */
+  def getIndexedBlockHeights(range: Range.Inclusive): DBIO[Seq[Int]] =
+    Tables.Blocks.filter(_.number between (range.start, range.end)).map(_.number).result
+
+  /**
+    * Get the latest block from the database.
+    */
+  def getLatestIndexedBlock: DBIO[Option[Tables.BlocksRow]] =
+    Tables.Blocks.sortBy(_.number.desc).take(1).result.headOption
+
+  /**
+    * Get a list of contract in a given block number range.
+    *
+    * @param range Inclusive range of the block's height
+    */
+  def getContracts(range: Range.Inclusive): DBIO[Seq[Tables.ContractsRow]] =
+    Tables.Contracts
+      .filter(_.blockNumber between (range.start, range.end))
+      .filter(c => c.isErc20 || c.isErc721)
```

Comment:
?

Reply:
It seems that the only usage is in `MantisOperations.scala` to `extractTokens` (which is why it filters Erc20/721 contracts), they name could be something more specific